### DISCUSSION TITLE
Vercel CLI as JSON output

### DISCUSSION THREAD

**Discussion by miklosme** [vercel/vercel#6704] [Help] [answered]
Is it possible to get the CLI result in JSON format?

I have to programmatically get the last deployment URL of a specific branch. I have the following command: `vercel list <project> --meta githubCommitRef=<branch>` which returns the correct URL. Of course, I could grab the URL with regexp, but is there a way to get the result in a machine parseable way?

If not, it would be nice to have a flag like `vercel list --json`

**Comment by ilyvion** [comment]
Wow. One and a half years, and not even a reply. Nice to know how serious they are about helping devs who want to use their tools programmatically.

**Comment by philwolstenholme** [comment, marked as answer]
Remember that there's also the Vercel REST API which returns JSON: https://vercel.com/docs/rest-api/endpoints

Not all the API endpoints are documented on that page, but you can look at the CLI code (e.g. [`cli/src/commands/rollback/request-rollback.ts`](https://github.com/vercel/vercel/blob/main/packages/cli/src/commands/rollback/request-rollback.ts)) to see what REST endpoints the CLI is calling.

I used this approach recently to work with deployment IDs and to add an instant rollback command to our GitLab CI pipeline at work: https://dev.to/philw_/using-vercels-instant-rollback-feature-in-your-own-cicd-pipeline-57oi
